infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -414,27 +409,26 @@ private void registerToLogReplicationLock() {
      * as source (sender/producer) or sink (receiver).
      */
     private void onLeadershipAcquire() {
-        switch (topologyDescriptor.getLocalClusterDescriptor().getRole()) {
-            case SOURCE:
-                log.info("Start as Source (sender/replicator)");
-                sessionManager.startReplication();
-                lockAcquireSample = recordLockAcquire(topologyDescriptor.getLocalClusterDescriptor().getRole());
-                processCountOnLockAcquire();
-                break;
-            case SINK:
-                log.info("Start as Sink (receiver)");
-
-                // Sink Site : the LogReplicationServer (server handler) will
-                // reset the LogReplicationSinkManager on acquiring leadership
-                interClusterServerNode.setLeadership(true);
-                lockAcquireSample = recordLockAcquire(topologyDescriptor.getLocalClusterDescriptor().getRole());
-
-                processCountOnLockAcquire();
-                break;
-            default:
-                log.error("Log Replication not started on this cluster. Leader node {} id {} belongs to cluster with {} role.",
-                        localEndpoint, localNodeId, topologyDescriptor.getLocalClusterDescriptor().getRole());
-                break;
+        // TODO [V2]: a cluster can be Source and/or Sink for different replication models. The support for this is
+        //  coming in the connectionModel PR
+        if (isSource()) {
+            log.info("Start as Source (sender/replicator)");
+            // TODO: Shama: currently the supported model is FULL_TABLE. So fetchConnectionEndpoints will get all SINKs.
```

Comment:
minor: should this comment be moved somewhere else (because fetchConnectionEndpoints is not here...)?